Extension/lib/filter/rules/csp-filter.js

```diff
@@ -19,6 +19,51 @@
     'use strict';
 
     /**
+     * Returns rule priority
+     * @param rule CSP rule
+     */
+    function getRulePriority(rule) {
+        if (!rule) {
+            return 0;
+        }
+        if (rule.whiteListRule && rule.isImportant) {
+            return 4;
+        } else if (rule.isImportant) {
+            return 3;
+        } else if (rule.whiteListRule) {
+            return 2;
+        }
+        return 1;
+    }
+
+    /**
+     * Decides which rule should be put into the given map.
+     * Compares priorities of the two given rules with the equal CSP directive and the rule that may already in the map.
+     *
+     * @param rule CSP rule (not null)
+     * @param whiteListRule CSP whitelist rule (may be null)
+     * @param map Rules mapped by csp directive
+     */
+    function putWithPriority(rule, whiteListRule, map) {
+
+        var cspDirective = rule.cspDirective;
+        var existRule = map[cspDirective];
+
+        var pr1 = getRulePriority(rule);
+        var pr2 = getRulePriority(whiteListRule);
+        var pr3 = getRulePriority(existRule);
+
+        var max = Math.max(pr1, pr2, pr3);
```

Comment:
If `pr1 === pr3` then rule1 is the same as rule3, so it doesn't matter which rule is used in the result.